processing/data_collection/gazette/spiders/sp_vinhedo.py

```diff
@@ -0,0 +1,56 @@
+from datetime import date, datetime
+import re
+
+from dateparser import parse
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpVinhedoSpider(BaseGazetteSpider):
+
+    # define class variables
+    allowed_domains = ['vinhedo.sp.gov.br']
+    name = 'sp_vinhedo'
+    start_urls = [
+        'http://vinhedo.sp.gov.br/arquivos/?wpdmc=boletim-municipal'
+    ]
+    start_date = date(2010, 12, 9)
+    TERRITORY_ID = '3556701'
+
+    # parse gazette pages dynamically
+    def parse(self, response):
+        pages = response.xpath('//*[@class="page-numbers"]//@href').getall()
+        last_page = re.search(r'(paged=)(\d{,2})', pages[-1]).group(2)
+        for i in range(1, int(last_page)+1):
+             url = f'{self.start_urls[0]}&paged={i}'
+             yield scrapy.Request(url, self.parse_page)
+
+    # parse gazette links per gazette page
+    def parse_page(self, response):
```

Comment:
**suggestion**: AS the URL in `start_urls` is already page 1, you could use the same `parse` method for all pages. No need to create a separated method. Just one is enough.